### PR TITLE
Status Chart: Daily cxx20/LWG stats, GraphQL rewrite

### PR DESCRIPTION
**Pull request by StephanTLavavej**
Gather daily stats for cxx20/lwg.
-----
* `gather_stats.js`
    + Change the beginning date to '2019-09-05', when all cxx20/lwg issues had been filed.
    + Use the same label criteria mentioned in the Explanation.
    + This improves the GitHub Issues line by excluding LWG Resolutions.
    + Process labels when transforming output. This avoids repeating work when analyzing each day.
* `daily_table.js`
    + Regenerate.
* `weekly_table.js`
    + Drop the cxx20/lwg stats starting at '2019-09-06'. (I verified that the lines were very close.)
    + 2020-07-24 update.
* `index.html`
    + Use `concat()` to fuse the old weekly and new daily stats.
    + Mention that Issues now excludes LWG.

GraphQL rewrite!
-----
* GitHub's [GraphQL API v4](https://docs.github.com/en/graphql) is faster and more powerful than their [REST API v3](https://docs.github.com/en/rest). This rewrite speeds up `gather_stats.js` by over 3.2x (22.8s => 7.0s, even faster when cached). It'll enable more detailed analysis in the near future. It also improves processing and progress display because PRs and issues are retrieved separately (and simultaneously), instead of being mixed together.
* `query_prs_and_issues.graphql` and `query_total_counts.graphql` are stored as separate files for ease of reading and formatting (with the Prettier extension).
* `package.json`: Change our dependencies, and configure Prettier.
* `gather_stats.js`
    + Format with Prettier.
    + Organize the code with functions. Note that `@octokit/graphql` is an async API.
    + This uses a `cliProgress.MultiBar` to display PR/issue progress, which corresponds to the queries we're sending. (The one with fewer items will finish first.)
    + GraphQL has a rate limit that can be queried in detail. Right now, it's nowhere close to being a problem, as our repo is small and our queries are extremely efficient. However, as our repo will steadily gain PRs/issues, and we'll perform more detailed analysis, the rate limit may be important in the future. So, I'm recording it and printing it at the end of the script.
    + This starts by running `query_total_counts.graphql` once, to get the total number of rate limit "points" (currently 5000), and the total numbers of PRs and issues (taking into account any deleted items, simplifying the previous logic). This could be fused into the later query, but at the cost of more complexity and more delay (as this initial query is small and fast, something like 200 ms).
    + Then, we use `query_prs_and_issues.graphql` to "paginate". This retrieves PRs and issues (currently in batches of 100 each, as specified by the query), and returns them as "nodes" containing just the fields we want. This is a major reason why it's so much more efficient - e.g. we are completely uninterested in PR/issue body text, which previously had to be retrieved from the database and sent over the network.
    + I'm using `try/finally` to deal with exceptions interacting with the progress bars.
    + The functions `transform_pr_nodes()` and `transform_issue_nodes()` closely correspond to the previous processing, except that the field names and structures are slightly different. Note that currently we don't use PR numbers for anything, and we use issue numbers only for warning, but I think it's a good idea to capture them for ease of debugging.
    + Note that we could paginate through labels too, but doing so would be more difficult. So, I'm simply using a hardcoded limit: our repo currently has a max of 4 labels per issue, so I've set the query to retrieve up to 5 labels and complain if it finds more. (Because this is multiplied by the number of issues being retrieved, asking for 100 labels would be expensive.)
    + `write_daily_table()` is also mostly unchanged, except that we can now iterate over PRs and issues separately.
    + At the end of the script, I print the rate limit info, and the total time taken.
    + I've changed the Luxon time diffing to use the default of milliseconds internally, since we always convert to our desired units at the end. This makes no difference for the generated file, it's just simpler to read.
* `daily_table.js`: Regen with GraphQL. Observe (in the isolated commit) that there are no differences except for additional days.

Live Preview: [stephantlavavej.github.io/STL/](https://stephantlavavej.github.io/STL/)
-----

### CODE REVIEW

**Comment by miscco** [comment]
That looks great. I would however, not use the combined age of PRs as that increases with the number of PRs. The average would be a better fit.

**Comment by StephanTLavavej** [comment]
Hmm, both metrics have value - earlier I thought that combined age was best if we could only have one, but I could easily generate average age too and make it dual-axis. (I also plan to generate “combined time since last review requested” and I suppose there could be an average counterpart too.) Will work on this for next week’s update, thanks!

File: query_prs_and_issues.graphql [hunk 1]
```diff
@@ -0,0 +1,38 @@
+query prs_and_issues($want_prs: Boolean!, $pr_cursor: String, $want_issues: Boolean!, $issue_cursor: String) {
```
Comment (CaseyCarter) [inline query_prs_and_issues.graphql:1]: So `pr_cursor` is sort of a meh `String`, but `want_prs` is very emphatically `Boolean`?

Comment (StephanTLavavej) [reply query_prs_and_issues.graphql:1]: Heh. This is how GraphQL marks required (`Meow!`) versus optional (`Meow`) parameters. The cursors are optional so they can be initially `null`, retrieving the first page of results.

**Pull request merged**